contributors/design-proposals/statefulset-update.md

```diff
@@ -108,29 +111,59 @@ applied to the StatefulSet.
+        // Type indicates the type of the StatefulSetUpdateStrategy.
+        Type StatefulSetUpdateStrategyType
+        // Partition is used to communicate the ordinal at which to partition 
+        // the StatefulSet when Type is PartitionedStatefulSetStrategyType. This 
+        // value must be set when Type is PartitionedStatefulSetStrategyType, 
+        // and it must be nil otherwise.
+        Partition *ParitionedStatefulSet
+}
+
+// StatefulSetUpdateStrategyType is a string enumeration type that enumerates 
+// all possible update strategies for the StatefulSet controller.
+type StatefulSetUpdateStrategyType string
+
+const (
+    // PartitionedStatefulSetStrategyType indicates that updates will only be 
+    // applied to a partition of the StatefulSet. This is useful for canaries 
+    // and phased roll outs.
+    PartitionedStatefulSetStrategyType StatefulSetUpdateStrategyType = "Partitioned"
+    // RollingUpdateStatefulSetStrategyType indicates that update will be 
+    // applied to all Pods in the StatefulSet with respect to the StatefulSet 
+    // ordering constraints.
+    RollingUpdateStatefulSetStrategyType = "RollingUpdate"
+    // OnDeleteStatefulSetStrategyType triggers the legacy behavior. Version 
+    // tracking and ordered rolling restarts are disabled. Pods are recreated 
+    // from the StatefulSetSpec when they are manually deleted.
+    OnDeleteStatefulSetStrategyType = "OnDelete"
+)
+
+// PartitionedStatefulSet contains the parameters used with the 
+// PartitionedStatefulSetStrategyType.
+type PartitionedStatefulSet struct {
+    // Ordinal indicates the ordinal at which the StatefulSet should be 
+    // partitioned.
+    Ordianl uint32
+}
+
 type StatefulSetSpec struct {
     // Replicas, Selector, Template, VolumeClaimsTemplate, and ServiceName 
     // omitted for brevity.
 
-	// TemplateGeneration is a monotonically increasing, 64 bit, integer used to 
-	// indicate the version of the of the PodTemplateSpec. If nil, the 
-	// StatefulSetController has not initialized its revision history,
-	// change tracking is not enabled, and all Pods will be created from 
-	// Template.
-	TemplateGeneration  *int64 `json:"templateGeneration"`
-
-	// GenerationPartition partitions the Pods in the StatefulSet by ordinal such 
-    // that all Pods with a lower ordinal will be created from the PodTemplate that
-    // represents the current revision of the StatefulSet's revision history and 
-    // all Pods with an a greater or equal ordinal will be created from the 
-    // PodTemplate that represents the target revision of the StatefulSet's 
-    // revision history.
-	GenerationPartition *int32 `json:"generationPartition,omitempty`
+	// UpdateStrategy indicates the StatefulSetUpdateStrategy that will be 
+	// employed to update Pods in the StatefulSet when a revision is made to 
+	// Template or VolumeClaimsTemplate.
+	UpdateStrategy StatefulSetUpdateStrategy `json:"updateStrategy,omitempty`
```

Comment:
Would we ever need a different strategy for claims vs pods . I guess we can think about it when we get there.